File: searching/linear_search/dart/linear_search.dart
void main() {
   linearSearch([1, 3, 4, 18, 9, 21], 3);
}

int linearSearch(List<int> nums, int items) {
  int numsLength = nums.length;
  int i;
  
  for (i = 0; i < numsLength; i++) {

    if(nums[i] == items) {
      print("Item is found at location $i");
      return i;
    }
  }
  
  i = -1;
   if(i == -1) {
    print("The item is not found");
  } 
  return i;
}

File: searching/linear_search/go/linear_search.go
package main

import "fmt"

func main() {

	nums := [...]int{1, 2, 21, 14, 27, 29, 11}
	var item int = 11
	var index = linearSearch(nums[:], item)
	if index != -1{
		fmt.Println(item, "Found at positions", index)
	} else {
		fmt.Println(item, "Not found")
	}
	
}

func linearSearch(nums []int, item int) int {
	n := len(nums)
	for i := 0; i < n; i++ {
		if nums[i] == item {
			return i
		}
	}
	return -1 
}
